src/display-buffer.coffee

```diff
@@ -332,6 +322,14 @@ class DisplayBuffer extends Model
     @scopedCharacterWidthsChangeCount++
     @characterWidthsChanged() unless @batchingCharacterMeasurement
 
+  setCharWidthsForLineId: (lineId, charWidths) ->
+    @charWidthsForLineId[lineId] = charWidths
+    @characterWidthsChanged() unless @batchingCharacterMeasurement
+
+  getCharWidthForLine: (line, index) ->
```

Comment:
:+1:

I'll do that once we agree on pushing this solution forward. What do you think?